hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/meta/CkpMetadata.java

```diff
@@ -20,14 +20,22 @@
 
 import org.apache.hudi.common.fs.FSUtils;
 import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.timeline.dto.CkpMetadataDTO;
```

Comment:
Can we extend a sub-class instead of messing up the existing logic then?

Reply:
> Can we extend a sub-class instead of messing up the existing logic then? 

Yes sure, already moved to a sub class. And I have refactor the constructors of `CkpMetadata`, now we have only one entry for creating ckpmetadata. 

> And do we have some fallback logic when the HTTP request timed out or throws ?

Yes, when requesting timeline server failed we will read from file system directly. Added tests about this.